pkg/scheduler/framework/plugins/noderesources/test_util.go

```diff
@@ -22,6 +22,83 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type containerResource struct {
+	cpu    int
+	memory int
+}
+
+func makePod(node string, milliCPU, memory int64) *v1.Pod {
+	return &v1.Pod{
+		Spec: v1.PodSpec{
+			NodeName: node,
+			Containers: []v1.Container{
+				{
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceCPU:    *resource.NewMilliQuantity(milliCPU, resource.DecimalSI),
+							v1.ResourceMemory: *resource.NewQuantity(memory, resource.DecimalSI),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func makePodWithContainers(node string, resources ...containerResource) *v1.Pod {
+	if len(resources) == 0 {
```

Comment:
this can be achieved by `st.MakePod().Node().Req().Req().Req()....` You can just pass-in variant resMap by changing the signature to `resMaps ...map[v1.ResourceName]string`.